### PR TITLE
feat(tier4_autoware_utils): add quaternion operation

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

motion_common is autoware.auto's unique implementation, where dependency from other packages will be removed soon. https://github.com/autowarefoundation/autoware.universe/issues/2229
In autoware.universe, we have other utils packages that are tier4_autoware_utils/motion_utils.

Some control packages depend on motion_common's implementation, which I copied them to tier4_autoware_utils in this PR.
https://github.com/autowarefoundation/autoware.universe/blob/4f6c3d4c5ae134985b2bc2ec0033e86a41530a0b/common/motion_common/src/motion_common/motion_common.cpp#L227-L252

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
This PR needs to be merged before https://github.com/autowarefoundation/autoware.universe/pull/2231 since control_performance_analysis uses this PR's functions.

File: common/tier4_autoware_utils/CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@ find_package(Boost REQUIRED)
 ament_auto_add_library(tier4_autoware_utils SHARED
   src/tier4_autoware_utils.cpp
   src/geometry/boost_polygon_utils.cpp
+  src/ros/operation.cpp
```
Comment (kenji-miyake) [inline common/tier4_autoware_utils/CMakeLists.txt:12]: I think it's good to make it more specific.
Or do you plan to add other operations to this file?

```suggestion
  src/ros/quaternion_operation.cpp
```

Comment (takayuki5168) [reply common/tier4_autoware_utils/CMakeLists.txt:12]: I think we can also add +, -, * of geometry_msgs's Pose and Point, whose operation is widely used in autoware.universe.

Comment (kenji-miyake) [reply common/tier4_autoware_utils/CMakeLists.txt:12]: How about `msg_operation` or `geometry_msg_operation`?

Comment (takayuki5168) [reply common/tier4_autoware_utils/CMakeLists.txt]: Sounds good to me.
https://github.com/autowarefoundation/autoware.universe/pull/2244/commits/6a310b7138c36c1d011a4458e1bf31dbc770d141

Comment (TakaHoribe) [reply common/tier4_autoware_utils/CMakeLists.txt]: Will this msg_operation support mathematical operations for other messages? like twist_a + twist_b

**Review by TakaHoribe — APPROVED** [review]
LGTM anyway. 
This is ported from motion_common, which was used in the autoware.auto

File: common/tier4_autoware_utils/CMakeLists.txt (hunk 1 above)
Comment (takayuki5168) [reply common/tier4_autoware_utils/CMakeLists.txt]: Of course we can add them if we want.
In this PR, I just copied from motion_common, which is currently required implementation for https://github.com/autowarefoundation/autoware.universe/pull/2244#issuecomment-1306697597

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**